nemo/collections/common/prompts/canary2.py

```diff
@@ -0,0 +1,206 @@
+    CANARY2_BOCTX,
+    CANARY_BOS,
+    CANARY_EOS,
+    CANARY_SPECIAL_TOKENIZER,
+)
+
+
+class Canary2PromptFormatter(PromptFormatter):
+    NAME = "canary2"
+    OUTPUT_ROLE = "assistant"
+    TEMPLATE = {
+        # User prompt.
+        # This is the main role used for training and ASR/AST inference.
+        "user": {
+            "template": f"{CANARY2_BOCTX}|decodercontext|{CANARY_BOS}|emotion||source_lang||target_lang||pnc||itn||timestamp||diarize|",
+            "slots": {
+                # Empty string or previous transcript / other context to bias predictions.
+                "decodercontext": Modality.Text,
+                # Emotion of the speaker - may be predicted by the model with a partial prompt.
+                "emotion": Modality.TextLiteral(
+                    "<|emo:undefined|>", "<|emo:neutral|>", "<|emo:angry|>", "<|emo:happy|>", "<|emo:sad|>"
+                ),
+                # Audio input language - may be predicted by the model with a partial prompt.
+                "source_lang": Modality.Text,
+                # Transcription language - specified by the user.
+                "target_lang": Modality.Text,
+                # Should we predict punctuation and capitalization?
+                "pnc": Modality.TextLiteral(
+                    "yes", "no", "true", "True", "false", "False", "1", "0", "pnc", "nopnc", "<|pnc|>", "<|nopnc|>"
+                ),
+                # Should we predict with inverse text normalization (numerals as digits, abbreviations, etc.)
+                "itn": Modality.TextLiteral(
+                    "yes", "no", "true", "True", "false", "False", "1", "0", "itn", "noitn", "<|itn|>", "<|noitn|>"
+                ),
+                # Should we predict timestamps?
+                "timestamp": Modality.TextLiteral(
+                    "yes",
+                    "no",
+                    "true",
+                    "True",
+                    "false",
+                    "False",
+                    "1",
+                    "0",
+                    "timestamp",
+                    "notimestamp",
+                    "<|timestamp|>",
+                    "<|notimestamp|>",
+                ),
+                # Should we diarize speech?
+                "diarize": Modality.TextLiteral(
+                    "yes",
+                    "no",
+                    "true",
+                    "True",
+                    "false",
+                    "False",
+                    "1",
+                    "0",
+                    "diarize",
+                    "nodiarize",
+                    "<|diarize|>",
+                    "<|nodiarize|>",
+                ),
+            },
+        },
+        # User prompt.
+        # This role is used for emotion / LID inference only - use it for two just two decoder inference steps
+        # to retrieve the recognized emotion and language tokens.
+        "user_partial": {
```

Comment:
Leaving a note that this doesn't cover the cases  like "identify language and transcribe with ITN" (which is fine for this version). should put sufficient guards regarding what users are allowed to ask in transcribe fn.

Reply:
That will be complicated to implement in transcribe fn in general. We should think about whether a new API specifically for Canary would be cleaner.

Reply:
+1 to new API, even already the prompt setup can be a bit bulky. Putting thought into API changes before it an actual problem could save some heartache.